module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -130,13 +130,100 @@ public function addCss($css)
     public function addJs($js)
     {
         if (!is_array($js) && !is_a($js, 'Traversable')) {
-            $js = [$js];
+            $this->addJsEntry($js);
+        } elseif (isset($js['file'])) {
+            $this->addJsEntry($js);
+        } elseif (isset($js[0])) {
```

Comment:
Have you tested what happens if you pass an empty array? That probably would never happen in reality, but if somebody had a JS-free theme, we should make sure that it doesn't break. I'd suggest that if the incoming value is empty, you return without doing anything rather than throwing an exception. It's worth adding this to the unit test.

Reply:
OK, i added support for this case & added a test case

Reply:
Thanks!